pkg/chrootarchive/archive_freebsd.go

```diff
@@ -0,0 +1,213 @@
+package chrootarchive // import "github.com/docker/docker/pkg/chrootarchive"
+
+import (
+	"bytes"
+	"encoding/gob"
+	"fmt"
+	"io"
+	"os"
+	"syscall"
+
+	"github.com/docker/docker/pkg/archive"
+	"github.com/docker/docker/pkg/reexec"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	packCmd        = "freebsd-pack-in-chroot"
+	unpackCmd      = "freebsd-unpack-in-chroot"
+	unpackLayerCmd = "freebsd-unpack-layer-in-chroot"
+)
+
+func init() {
+	reexec.Register(packCmd, reexecMain(packInChroot))
+	reexec.Register(unpackCmd, reexecMain(unpackInChroot))
+	reexec.Register(unpackLayerCmd, reexecMain(unpackLayerInChroot))
+}
+
+func reexecMain(f func(options archive.TarOptions, args ...string) error) func() {
+	return func() {
+		options, err := recvOptions()
+		root := os.Args[1]
+
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+
+		if err := syscall.Chroot(root); err != nil {
+			fmt.Fprintf(
+				os.Stderr,
+				"could not chroot: %v\n",
+				os.PathError{Op: "chroot", Path: root, Err: err},
+			)
+			os.Exit(1)
```

Comment:
Can you increment the exit codes here and at L50?
This way exit code 1 is caller error, 2 is chroot error, and 3 is operation error.

Reply:
Makes sense, thanks!